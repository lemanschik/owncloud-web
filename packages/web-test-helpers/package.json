{
  "name": "web-test-helpers",
  "private": true,
  "main": "src/index.ts",
  "peerDependencies": {
    "@casl/ability": "^6.3.3",
    "@casl/vue": "^2.2.1",
    "axios": "0.27.2",
    "design-system": "@ownclouders/design-system",
    "vue3-gettext": "^2.3.3",
    "vue-router": "4.1.6",
    "vuex": "4.1.0"
  }
}
