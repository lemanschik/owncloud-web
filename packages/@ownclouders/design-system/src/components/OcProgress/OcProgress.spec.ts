import { shallowMount } from 'web-test-helpers'
import Progress from './OcProgress.vue'

describe('OcProgress', () => {
  it('sets correct classes', () => {
    const wrapper = shallowMount(Progress, {
      props: {
        value: 3,
        max: 10,
        variation: 'warning',
        size: 'small'
      }
    })

    expect(wrapper.classes()).toContain('oc-progress-small')
    expect(wrapper.classes()).toContain('oc-progress-warning')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
