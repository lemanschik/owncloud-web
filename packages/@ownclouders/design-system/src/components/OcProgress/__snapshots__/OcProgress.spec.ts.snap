// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcProgress sets correct classes 1`] = `
<div aria-busy="true" aria-valuemax="10" aria-valuemin="0" aria-valuenow="3" class="oc-progress oc-progress-small oc-progress-warning" role="progressbar">
  <div class="oc-progress-current" style="width: 30%;"></div>
</div>
`;
